tools/meson-build.sh

```diff
@@ -10,7 +10,7 @@ CC="$5"
 CXX="$6"
 
 # shellcheck disable=SC2086
-[ -f "$dst/ninja.build" ] || CC="$CC" CXX="$CXX" meson "$src" "$dst" $options
+[ -f "$dst/.ninja_deps" ] || CC="$CC" CXX="$CXX" meson "$src" "$dst" $options
```

Comment:
So I'm pretty sure this changed in meson (or ninja?). No idea when. Should we support both?

Reply:
If I read the script right, it wants to run `meson setup` only if the build directory hasn't been setup yet. The `ninja` build rules are in `ninja.build` thus it seems to be one correct candidate for establishing that. `.ninja_deps` stores the dependencies between targets. It should not be present if `ninja` has not be executed at least once.

You may want to make the setup step explicit invoking `meson setup "$src" "$dst" $options`. I think recent version of meson even warn about this.

If `$options` can change, you may want to run `meson setup --reconfigure "$src" "$dst" $options` instead of skipping the `setup` steps. In this way the new options take effect.